gcc/config/riscv/riscv.cc

```diff
@@ -2070,9 +2064,11 @@ riscv_v_adjust_scalable_frame (rtx target, poly_int64 offset, bool epilogue)
 /* Check if post inc instructions are valid. If not, make the address
  * vaild. */
 bool
-riscv_legitimate_post_inc_p (machine_mode mode, rtx x, bool strict_p)
+riscv_legitimate_post_inc_p (rtx x)
```

Comment:
The naming of this function is misleading because it returns true for non-post-increment base+reg addresses as well as post-increment address. Renaming the function to `riscv_legitimate_xcvmem_address_p` would more accurately describe its purpose.